libraries/core_libs/network/src/tarcap/packets_handlers/status_packet_handler.cpp

```diff
@@ -105,49 +106,52 @@ void StatusPacketHandler::process(const PacketData& packet_data, const std::shar
     }
 
     auto it = packet_data.rlp_.begin();
-    selected_peer->dag_level_ = (*it++).toPositiveInt64();
-    selected_peer->pbft_chain_size_ = (*it++).toPositiveInt64();
+    selected_peer->dag_level_ = (*it++).toInt<uint64_t>();
+    selected_peer->pbft_chain_size_ = (*it++).toInt<uint64_t>();
     selected_peer->syncing_ = (*it++).toInt();
-    selected_peer->pbft_round_ = (*it++).toPositiveInt64();
+    selected_peer->pbft_round_ = (*it++).toInt<uint64_t>();
     selected_peer->pbft_previous_round_next_votes_size_ = (*it++).toInt<unsigned>();
 
+    // TODO: Address malicious status
+    if (!syncing_state_->is_pbft_syncing()) {
+      if (pbft_synced_period < selected_peer->pbft_chain_size_) {
+        LOG(log_nf_) << "Restart PBFT chain syncing. Own synced PBFT at period " << pbft_synced_period
+                     << ", peer PBFT chain size " << selected_peer->pbft_chain_size_;
+        if (pbft_synced_period < selected_peer->pbft_chain_size_ + 1) {
```

Comment:
Dose this should be "(pbft_synced_period + 1) < selected_peer->pbft_chain_size_"?

Reply:
Yes. Good catch. I fixed it